rest-service/manager_rest/resource_manager.py

```diff
@@ -2676,140 +2617,76 @@ def _get_utc_datetime_from_sched_plan(time_expression, base_datetime=None):
-        for parent_id, child_ids in children.items():
-            for child_id in child_ids:
-                parents.setdefault(child_id, set()).add(parent_id)
-
-        no_children = {dep_id for dep_id in deps if not children.get(dep_id)}
-
-        # we've not fetched any children of those, so we just take whatever
-        # counts they declare
-        for dep_id in no_children:
-            summary[dep_id] += deps[dep_id]
-
-        while no_children:
-            dep_id = no_children.pop()
-            dep = deps[dep_id]
-            if dep.is_env:
-                child_summary = _SubDepSummary(
-                    sub_environments_count=1,
-                    sub_environments_status=dep.deployment_status
-                )
-            else:
-                child_summary = _SubDepSummary(
-                    sub_services_count=1,
-                    sub_services_status=dep.deployment_status
-                )
-            for parent_id in parents.get(dep_id, set()):
-                summary[parent_id] += child_summary + summary[dep_id]
-                children[parent_id].remove(dep_id)
-                if not children[parent_id]:
-                    no_children.add(parent_id)
-        return summary
-
     def recalc_ancestors(self, deployment_ids):
         """Recalculate statuses & counts for all ancestors of deployment_ids"""
         if not deployment_ids:
             return
-        tree = models.DeploymentLabelsDependencies._get_update_tree(
-            deployment_ids)
-        if not tree:
-            return self._reset_sub_deployments(deployment_ids)
-        deps, children = self._structure_update_tree(tree)
-        summary = self._summarize_tree(deps, children)
-        for dep_id, dep_summary in summary.items():
-            dep = deps[dep_id]
-            new_status = models.Deployment.decide_deployment_status(
-                latest_execution_status=dep.latest_execution_status,
-                installation_status=dep.installation_status,
-                sub_services_status=dep_summary.sub_services_status,
-                sub_environments_status=dep_summary.sub_environments_status
-            )
-            db.session.execute(
-                models.Deployment.__table__.update()
-                .where(models.Deployment.__table__.c._storage_id == dep_id)
-                .values(
-                    deployment_status=new_status,
-                    sub_services_count=dep_summary.sub_services_count,
-                    sub_environments_count=dep_summary.sub_environments_count,
-                    sub_services_status=dep_summary.sub_services_status,
-                    sub_environments_status=dep_summary.sub_environments_status
+        with self.sm.transaction():
+            deps = models.DeploymentLabelsDependencies.get_dependencies(
+                deployment_ids, dependents=False, locking=True)
+            if not deps:
+                # no deps, means there's no tree to speak of, we just need to
```

Comment:
This should be "we only need to update deployment_ids" or "we just need to update deployment_ids", but I'm just commenting on it in case it wasn't a braino- I don't think it's necessarily worth fixing for this PR.

Reply:
`we just need to update deployment_ids only` is the wording i changed this to, for extra emphasis